Documentation/check-links.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+build_dir="${script_dir}/_build"
+warnings="${build_dir}/links_warnings.txt"
+linksdir="${build_dir}/links"
+
+cd "${script_dir}"
+mkdir -p "${build_dir}"
+rm -f -- "${warnings}"
+
+filter_warnings() {
+    [ -s "${warnings}" ] || return
+    grep -v -E \
+        -e "/.*/:[0-9]+: WARNING: circular inclusion in \"include\" directive"
+        "${warnings}"
+}
+
+has_build_warnings() {
+    filter_warnings > /dev/null
+}
+
+build_with_linkchecker() {
+    set +o nounset
+    if [[ -n "$MAKE_GIT_REPO_SAFE" ]]; then
+        git config --global --add safe.directory "${root_dir}"
+    fi
+    set -o nounset
```

Comment:
The spell-checker needed this part because it runs some Git commands (to retrieve the list of authors and consider them as spelling exceptions), but I don't expect the linkchecker to do that, so we can probably remove this bit.